packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -2,24 +2,30 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { useSelect } from '@wordpress/data';
-import { useState, useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
 import {
 	Tooltip,
 	DropdownMenu,
 	MenuGroup,
 	MenuItemsChoice,
 	MenuItem,
 } from '@wordpress/components';
-import { Icon, home, plus } from '@wordpress/icons';
+import { Icon, home, plus, undo } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import AddTemplate from '../add-template';
 import TemplatePreview from './template-preview';
 import ThemePreview from './theme-preview';
 
+const TEMPLATE_OVERRIDES = {
```

Comment:
What's the path forward for supporting more of the possible overrides in the template hierarchy? Will we just add more keys here, or make it more dynamic?

Reply:
Both are viable. It depends on what templates we need to support.